vcd/catalogitem.go

```diff
@@ -46,7 +46,16 @@ func deleteCatalogItem(d *schema.ResourceData, vcdClient *VCDClient) error {
 	return nil
 }
 
-// Finds catalog item which can be vapp template OVA or media ISO file
+// vappTemplateToCatalogItem returns the catalog item corresponding to the associated vApp template
+func vappTemplateToCatalogItem(vappTemplateName string, catalog *govcd.Catalog) (*govcd.CatalogItem, error) {
+	item, err := catalog.GetCatalogItemByName(vappTemplateName, false)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving catalog item from vapp template %s: %s", vappTemplateName, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error retrieving catalog item from vApp template %s: %s", vappTemplateName, err)
```

Reply:
removed this function, as not needed after refactoring